filebeat/harvester/log.go

```diff
@@ -44,6 +44,29 @@ func (h *Harvester) Harvest() {
 		return
 	}
 
+	go func() {
+		select {
+		case <-h.prospectorDone:
+			close(h.done)
+		case <-h.done:
+			return
+		}
+
+	}()
+
+	if h.config.CloseTimeout > 0 {
```

Comment:
config requires CloseTimeout to be `min=0,nonzero`. This makes `0` an invalid config file option. With default being 0, it can still be 0. Reminds me, ucfg should validate defaults, if value is not configured.